pyproject.toml

```diff
@@ -8,6 +8,7 @@ build = "build.py"
 
 [tool.poetry.dependencies]
 python = "^3.9"
+progress = "^1.6"
```

Comment:
I actually don't want any third-party dependencies. Since `progress` is a quite lightweight project, can you just migrate to `chaine`? As a submodule?

Reply:
A refactoring of `progress` is necessary in the first place, because we don't need all these different stylings of the progress bar:

![68747470733a2f2f7261772e6769746875622e636f6d2f7665726967616b2f70726f67726573732f6d61737465722f64656d6f2e676966](https://user-images.githubusercontent.com/16133277/181745567-7505bfa8-1708-40e3-b5c2-7c447167d200.gif)


Just a single, simple and plain progress bar.

Reply:
I'd prefer the `Bar` one

Reply:
I also prefer the `Bar` one, the integration of `progress` is currently temporary and for testing purposes. The main reason for this very drafty PR is, that I'm continue to work on this PR on another computer.

I'm currently not sure if I will integrate this directly into the C code or as an outer layer within the Python code, but I would prefer the first one to have more control over the bar.

Reply:
Nice: https://github.com/doches/progressbar